### PR TITLE
Support for secure / kerberized clusters

### PR DESCRIPTION
**Pull request by laserson**
This patch makes our test suite pass against a kerberized cluster.  cc @caseyching for review.

Related: #366, #406

### CODE REVIEW

**Comment by laserson** [comment]
@wesm I added a smidge of docs in this patch as well.

**Comment by wesm** [comment]
btw no need to squash your own commits; let's use `dev/merge-pr.py` to do it

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -3,3 +3,30 @@
...
+Therefore, the connection semantics are similar to the other access methods for
+working with secure clusters.
+
+Specifically, after authenticating yourself against Kerberos (e.g., by issuing
+the appropriate ``kinit`` commmand), simply pass ``use_kerberos=True`` (and set
+``kerberos_service_name`` if necessary) to the ``ibis.impala_connect(...)``
+method when instantiating an ``ImpalaConnection``.  This method also takes
+arguments to configure LDAP (``use_ldap``, ``ldap_user``, and
+``ldap_password``) and SSL (``use_ssl``, ``ca_cert``).  See the documentation
+for the Impala shell for more details.
```
Comment (wesm) [inline docs/source/configuration.rst]: Maybe provide sample code?

File: ibis/__init__.py [hunk 2]
```diff
@@ -99,7 +99,8 @@ def impala_connect(host='localhost', port=21050, protocol='hiveserver2',
     return ImpalaConnection(pool_size=pool_size, **params)
 
 
-def hdfs_connect(host='localhost', port=50070, protocol='webhdfs', **kwds):
+def hdfs_connect(host='localhost', port=50070, protocol='webhdfs',
+                 use_kerberos=False, verify=True, **kwds):
```
Comment (wesm) [inline ibis/__init__.py]: can you add document these in the docstring?

File: docs/source/configuration.rst (hunk 1 above)
Comment (laserson) [inline docs/source/configuration.rst]: meh.  It's literally the same except for `use_kerberos=True`.  This'll only increase the places we need to make changes.  I instead added a note to the main getting-started doc.

File: ibis/__init__.py (hunk 2 above)
Comment (laserson) [inline ibis/__init__.py]: done.

**Comment by wesm** [comment]
lgtm. i will merge here shortly

**Pull request closed without merging**